compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -703,6 +703,144 @@ OMR::ARM64::TreeEvaluator::vmulEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    return inlineVectorBinaryOp(node, cg, mulOp);
    }
 
+TR::Register *
+OMR::ARM64::TreeEvaluator::vdivIntHelper(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   struct DivOps {
+      TR::InstOpCode::Mnemonic vectorToGPROp;
+      TR::InstOpCode::Mnemonic divOp;
+      TR::InstOpCode::Mnemonic gprToVectorOp;
+      uint32_t                 numElements;
+   };
+   static const struct DivOps ops[4] = {
+      {TR::InstOpCode::smovwb, TR::InstOpCode::sdivw, TR::InstOpCode::vinswb, 16},
+      {TR::InstOpCode::smovwh, TR::InstOpCode::sdivw, TR::InstOpCode::vinswh, 8},
+      {TR::InstOpCode::umovws, TR::InstOpCode::sdivw, TR::InstOpCode::vinsws, 4},
+      {TR::InstOpCode::umovxd, TR::InstOpCode::sdivx, TR::InstOpCode::vinsxd, 2}
+   };
+   TR::DataType eType = node->getDataType().getVectorElementType();
+   const uint32_t index = (eType == TR::Int8) ? 0 :
+                         ((eType == TR::Int16) ? 1 :
+                         ((eType == TR::Int32) ? 2 : 3));
+   struct DivOps op = ops[index];
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *lhsReg = cg->evaluate(firstChild);
+   TR::Register *rhsReg = cg->evaluate(secondChild);
+
+   TR::Register *resultReg = cg->allocateRegister(TR_VRF);
+   TR_ARM64ScratchRegisterManager *srm = cg->generateScratchRegisterManager();
+   TR::Register *tmp1Reg = srm->findOrCreateScratchRegister(TR_GPR);
+   TR::Register *tmp2Reg = srm->findOrCreateScratchRegister(TR_GPR);
+
+   const uint32_t numElementsPerLoop = (op.numElements > 4) ? 4 : op.numElements;
+   const uint32_t loopCount = op.numElements / numElementsPerLoop;
+
+  /*
+   * For 8-bit elements,
+   *
+   * movzw  w2, #4
+   * loop:
+   * vushr4s v2, v2, #8
+   * smovwb w0, v0.b[0]
+   * smovwb w1, v1.b[0]
+   * sdivw  w0, w1, w0
+   * vinswb v2.b[3], w0
+   * smovwb w0, v0.b[4]
+   * smovwb w1, v1.b[4]
+   * sdivw  w0, w1, w0
+   * vinswb v2.b[7], w0
+   * smovwb w0, v0.b[8]
+   * smovwb w1, v1.b[8]
+   * sdivw  w0, w1, w0
+   * vinswb v2.b[11], w0
+   * smovwb w0, v0.b[12]
+   * smovwb w1, v1.b[12]
+   * sdivw  w0, w1, w0
+   * vinswb v2.b[15], w0
+   * vushr4s v0, v0, #8
+   * vushr4s v1, v1, #8
+   * subw  w2, w2, #1
+   * cbnzw w2, loop
```

Comment:
I would suggest aligning the operands, or always using a single space between the mnemonic and the operands:

```
   * vushr4s v1, v1, #8
   * subw    w2, w2, #1
   * cbnzw   w2, loop
```

```
   * vushr4s v1, v1, #8
   * subw w2, w2, #1
   * cbnzw w2, loop
```

Reply:
Fixed the alignment.